### PR TITLE
HADOOP-16536. Backport HADOOP-15273 to branch-2

### PR DESCRIPTION
**Pull request by kai33**
[HADOOP-16536](https://jira.apache.org/jira/browse/HADOOP-16536)
[HADOOP-15273](https://jira.apache.org/jira/browse/HADOOP-15273)

this PR is to backport HADOOP-15273 first.
Once it's merged, I'll submit another one for HADOOP-16158.

### CODE REVIEW

**Comment by jojochuang** [comment]
Can you raise one PR for HADOOP-15273 after HADOOP-16158 (or the other way around)?  I'd rather not to squash these commits into one.

**Comment by kai33** [comment]
Sure, I'll split it to separate ones

**Comment by kai33** [comment]
Update this PR to backport HADOOP-15273 first.
Once it's merged, I'll submit another one for HADOOP-16158.

File: hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java [hunk 1]
```diff
@@ -207,15 +207,30 @@ private void compareCheckSums(FileSystem sourceFS, Path source,
...
-          .append(source).append(" and ").append(target).append(".");
-      if (sourceFS.getFileStatus(source).getBlockSize() !=
+      StringBuilder errorMessage =
+          new StringBuilder("Checksum mismatch between ")
+              .append(source).append(" and ").append(target).append(".");
+      boolean addSkipHint = false;
+      String srcScheme = sourceFS.getScheme();
+      String targetScheme = targetFS.getScheme();
+      if (!srcScheme.equals(targetScheme)
+          && !(srcScheme.contains("hdfs") && targetScheme.contains("hdfs"))) {
```
Comment (kai33) [inline hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java]: This check will be removed once HADOOP-16158 is backported

**Pull request closed without merging**